soh/src/overlays/misc/ovl_kaleido_scope/z_kaleido_item.c

```diff
@@ -598,223 +613,35 @@ void KaleidoScope_UpdateItemEquip(GlobalContext* globalCtx) {
 
             osSyncPrintf("\n＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝\n");
 
-            if (pauseCtx->equipTargetCBtn == 0) {
```

Comment:
This chunk of code is a rewrite of the code that handles assigning the selected item in the item menu to the chosen C button, and also swapping it in the case the item is already on another C button. First of all, the code includes checks that I believe don't apply to the final game (The checks for `(pauseCtx->equipTargetItem >= 0xBF) && (pauseCtx->equipTargetItem <= 0xC1)`) and secondly, enumurate every possible case instead of doing it in a loop. I opted to completely rewrite the code to loop over the other buttons and swap if needed.